### PR TITLE
Add manual ON/OFF on device page

### LINKED ISSUE #11: Add manual ON/OFF on device page
- [x] runner+ level
- [x] log every action with username and run_option

### PR DESCRIPTION
**Pull request by just806me**
resolves #11

### CODE REVIEW

File: app/web/controllers/audio_files.py [hunk 1]
```diff
@@ -14,6 +17,37 @@
...
+@dataclass
+class FileEntry:
+    name: str
+    prefix: str
+    file_callback: Optional[str]
+
+    @property
+    def url(self) -> Optional[str]:
+        if self.file_callback is not None:
+            file_opt = self.name if self.prefix == '' else f"{self.prefix}/{self.name}"
```
Comment (just806me) [inline app/web/controllers/audio_files.py:42]: to be the same as line 28

```suggestion
            file_opt = f'{self.prefix}/{self.name}' if self.prefix else self.name
```

File: app/web/controllers/commands.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    def index(self, device_id: Optional[str] = None, file_opt: Optional[str] = None,
+              success: Optional[str] = None, error: Optional[str] = None,
+              run_options: Optional[str] = ""):
+        devices = Device.enabled()
+        if file_opt is not None:
+            run_options = f"file: '{file_opt}'"
+
+        selected_device = None
+        if device_id is not None:
+            selected_device = Device.find(int(device_id))
```
Comment (just806me) [inline app/web/controllers/commands.py:32]:
```suggestion
        selected_device = Device.find(int(device_id)) if device_id else None
```

File: app/web/www/audio_files/index.html [hunk 3]
```diff
@@ -9,30 +9,38 @@
                 <div class="eight wide column">
                     <h2 class="ui header">Присутні файли</h2>
                     <div class="ui list">
-                        {% if back_prefix is not none %}
-                            <a class="item" href="/audio_files?prefix={{ back_prefix }}">
+                        {% if back_url is not none %}
+                            <a class="item" href="{{back_url}}">
```
Comment (just806me) [inline app/web/www/audio_files/index.html:13]:
```suggestion
                            <a class="item" href="{{ back_url }}">
```

File: app/web/www/audio_files/index.html [hunk 4]
```diff
@@ -9,30 +9,38 @@
...
                                 {{ dir.name }}
                             </a>
                         {% endfor %}
                         {% for file in files %}
-                            <div class="item">
-                                <i class="file icon"></i>
-                                {{ file.name }}
-                            </div>
+                            {% if file.has_url %}
+                                <a class="item" href="{{file.url}}">
```
Comment (just806me) [inline app/web/www/audio_files/index.html:26]:
```suggestion
                                <a class="item" href="{{ file.url }}">
```

File: app/web/www/commands/index.html [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                                <option value="device_start">Запуск</option>
+                                <option value="device_stop">Зупинка</option>
+                            </select>
+                        </div>
+                        <div class="field">
+                            <label>Пристрій</label>
+                            <select class="ui fluid dropdown" name="device_id" required>
+                                <option value=""></option>
+                                {% for dev in devices %}
+                                    <option {{ 'selected' if selected_device.id == dev.id }} value="{{dev.id}}">{{dev.name}}</option>
```
Comment (just806me) [inline app/web/www/commands/index.html:35]:
```suggestion
                                    <option {{ 'selected' if selected_device is not none and selected_device.id == dev.id }} value="{{ dev.id }}">{{ dev.name }}</option>
```

File: app/web/www/commands/index.html [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                                {% endfor %}
+                            </select>
+                        </div>
+                        <div class="field">
+                            <label>Параметри</label>
+                            <textarea rows="2" name="run_options">{{ run_options if run_options != ""}}</textarea>
```
Comment (just806me) [inline app/web/www/commands/index.html:41]: is the condition required?

```suggestion
                            <textarea rows="2" name="run_options">{{ run_options }}</textarea>
```

File: app/web/controllers/audio_files.py [hunk 7]
```diff
@@ -28,28 +62,34 @@ def __handle_prefix(self, prefix: str) -> tuple[list[str], Path]:
...
+        query = to_query_string({'prefix': back_prefix, 'file_callback': file_callback})
+        back_url = f"/audio_files?{query}"
+
+        return self.template.render({'dirs': dirs, 'files': files, 'back_url': back_url,
+                                     'prefix': prefix, 'file_callback': file_callback})
 
     @cherrypy.expose
     @cherrypy.tools.allow(methods=['POST'])
     @cherrypy.tools.authenticate()
     @cherrypy.tools.authorize(role=UserRole.MODERATOR)
```
Comment (just806me) [inline app/web/controllers/audio_files.py:90]: Should the role be also updated here?

File: app/web/controllers/audio_files.py [hunk 8]
```diff
@@ -28,28 +62,34 @@ def __handle_prefix(self, prefix: str) -> tuple[list[str], Path]:
     @cherrypy.expose
     @cherrypy.tools.allow(methods=['GET'])
     @cherrypy.tools.authenticate()
-    @cherrypy.tools.authorize(role=UserRole.MODERATOR)
-    def index(self, prefix: str = ''):
+    @cherrypy.tools.authorize(role=UserRole.RUNNER)
```
Comment (just806me) [inline app/web/controllers/audio_files.py:65]: Should the role be also updated for upload_files and create_dir?

File: app/web/controllers/audio_files.py [hunk 9]
```diff
@@ -28,28 +62,34 @@ def __handle_prefix(self, prefix: str) -> tuple[list[str], Path]:
...
+    def index(self, prefix: str = '', file_callback: Optional[str] = None):
```
Comment (just806me) [inline app/web/controllers/audio_files.py:66]: Should the role be also updated for `upload_files` and `create_dir`?

Comment (kostyanf14) [reply app/web/controllers/audio_files.py:66]: For now I think no

**Review by kostyanf14 — APPROVED** [review]

**Pull request merged**